OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -62,6 +62,10 @@ enum ChatPanelType { Lobby, Global }
 		readonly LabelWidget chatLabel;
 		bool teamChat;
 
+		int lobbyChatUnreadMessages;
+		int globalChatLastReadMessages;
```

Comment:
Making this `static` fixes the thing that RoosterDragon mentioned. I think I'd like it better that way, too.

You just need a way to reset it to 0 when you press the `leave chat` button.